taichi/backends/cuda/cuda_profiler.cpp

```diff
@@ -0,0 +1,150 @@
+#include "taichi/backends/cuda/cuda_profiler.h"
+#include "taichi/backends/cuda/cuda_driver.h"
+#include "taichi/backends/cuda/cuda_context.h"
+
+TLANG_NAMESPACE_BEGIN
+
+CUDAProfiler::CUDAProfiler() {
+  TI_TRACE("CUDAProfiler::CUDAProfiler() ");
+}
+
+CUDAProfiler::~CUDAProfiler() {
+  if (profiler_config_.profiler_type == CUDA_KERNEL_PROFILER_CUPTI) {
+    end_profiling();
+    deinit_cupti();
+  }
+}
+
+bool CUDAProfiler::set_profiling_mode(KernelProfilerMode profiling_mode) {
+  bool enable = profiling_mode == KernelProfilerMode::enable |
```

Comment:
why not just `!= KernelProfilerMode::disable`?

Reply:
thanks, I encapsulated this logic into is_cuda_profiler() and simplified init logic in commit e05ee990e753783bab8d57c1638e5d78893675f6